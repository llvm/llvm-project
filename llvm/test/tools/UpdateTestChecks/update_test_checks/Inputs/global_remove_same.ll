; RUN: opt -S < %s | FileCheck %s

define i32 @foo() {
; CHECK-LABEL: @foo(
; CHECK-NEXT:    [[RESULT:%.*]] = call i32 @bar(i32 0, i32 1)
; CHECK-NEXT:    ret i32 [[RESULT]]
;
  %result = call i32 @bar(i32 0, i32 1)
  ret i32 %result
}

declare i32 @bar(i32, i32)
; CHECK-LABEL: @bar(
; CHECK-SAME: i32
; CHECK-SAME: i32
