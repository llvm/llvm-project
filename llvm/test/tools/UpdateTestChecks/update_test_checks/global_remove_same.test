## Basic test checking global checks split over multiple lines are removed together
# RUN: cp -f %S/Inputs/global_remove_same.ll %t.ll && %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/global_remove_same.ll.expected

