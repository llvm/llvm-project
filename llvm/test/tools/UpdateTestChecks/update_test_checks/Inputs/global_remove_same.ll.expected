; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
; RUN: opt -S < %s | FileCheck %s

define i32 @foo() {
; CHECK-LABEL: @foo(
; CHECK-NEXT:    [[RESULT:%.*]] = call i32 @bar(i32 0, i32 1)
; CHECK-NEXT:    ret i32 [[RESULT]]
;
  %result = call i32 @bar(i32 0, i32 1)
  ret i32 %result
}

declare i32 @bar(i32, i32)
